packages/zoe/test/unitTests/contracts/test-otcDesk.js

```diff
@@ -0,0 +1,359 @@
+
+const root = `${__dirname}/../../../src/contracts/otcDesk`;
+
+const installCode = async zoe => {
+  const bundle = await bundleSource(root);
+  const installation = await E(zoe).install(bundle);
+  return installation;
+};
+
+const makeAlice = async (t, zoe, installation, issuers, origPayments) => {
+  let creatorFacet;
+  const {
+    moolaIssuer,
+    simoleanIssuer,
+    bucksIssuer,
+    moola,
+    simoleans,
+    bucks,
+  } = issuers;
+  const { moolaPayment, simoleanPayment, bucksPayment } = origPayments;
+  const simoleanPurse = await E(simoleanIssuer).makeEmptyPurse();
+  return {
+    installCode: async () => {},
+    startInstance: async () => {
+      ({ creatorFacet } = await E(zoe).startInstance(installation));
+    },
+    addInventory: async () => {
+      const invitation = await E(creatorFacet).makeAddInventoryInvitation({
+        Moola: moolaIssuer,
+        Simolean: simoleanIssuer,
+        Buck: bucksIssuer,
+      });
+      const proposal = harden({
+        give: {
+          Moola: moola(10000),
+          Simolean: simoleans(10000),
+          Buck: bucks(10000),
+        },
+      });
+      const payments = {
+        Moola: moolaPayment,
+        Simolean: simoleanPayment,
+        Buck: bucksPayment,
+      };
+
+      const seat = await E(zoe).offer(invitation, proposal, payments);
+      const offerResult = await E(seat).getOfferResult();
+      t.is(offerResult, 'Inventory added');
+    },
+    makeQuoteForBob: async (userGiveQuote, userWantQuote, timer, deadline) => {
+      const bobInvitation = await E(creatorFacet).makeQuote(
+        userGiveQuote,
+        userWantQuote,
+        timer,
+        deadline,
+      );
+      return bobInvitation;
+    },
+    removeInventory: async () => {
+      const invitation = await E(creatorFacet).makeRemoveInventoryInvitation();
+      // Intentionally do not remove it all
+      const proposal = harden({ want: { Simolean: simoleans(2) } });
+      const seat = await E(zoe).offer(invitation, proposal);
+      const offerResult = await E(seat).getOfferResult();
+      t.is(offerResult, 'Inventory removed');
+      const simoleanPayout = await E(seat).getPayout('Simolean');
+      const amountDeposited = await simoleanPurse.deposit(simoleanPayout);
+      t.deepEqual(amountDeposited, simoleans(2));
+    },
+  };
+};
+
+const makeBob = (t, zoe, installation, issuers, origPayments) => {
+  const {
+    moolaIssuer,
+    simoleanIssuer,
+    bucksIssuer,
+    moola,
+    simoleans,
+    bucks,
+  } = issuers;
+  const { moolaPayment, simoleanPayment, bucksPayment } = origPayments;
+  const moolaPurse = moolaIssuer.makeEmptyPurse();
+  const simoleanPurse = simoleanIssuer.makeEmptyPurse();
+  const bucksPurse = bucksIssuer.makeEmptyPurse();
+
+  moolaPurse.deposit(moolaPayment);
+  simoleanPurse.deposit(simoleanPayment);
+  bucksPurse.deposit(bucksPayment);
+  return harden({
+    offerGood: async untrustedInvitation => {
```

Comment:
This name seems backwards. Bob's method should reflect that he's receiving an invitation

Reply:
Well, it's the name of what action Bob is taking. In this case he is offering without triggering an error (offer - good). I can see why the name isn't great because goods are also a thing.